plugins/module_utils/network/iosxr/config/bgp_global/bgp_global.py

```diff
@@ -0,0 +1,421 @@
+            "bgp_origin_as_validation_time_off",
+            "bgp_origin_as_validation_time",
+            "bgp_default_information_originate",
+            "bgp_default_metric",
+            "bgp_graceful_maintenance",
+            "ibgp_policy_out_enforce_modifications",
+            "mpls_activate_interface",
+            "mvpn",
+            "nsr_set",
+            "nsr_disable",
+            "socket_receive_buffer_size",
+            "socket_send_buffer_size",
+            "update_in_error_handling_basic_ebgp_disable",
+            "update_in_error_handling_basic_ibgp_disable",
+            "update_in_error_handling_extended_ebgp",
+            "update_in_error_handling_extended_ibgp",
+            "update_out_logging",
+            "update_limit",
+            "rpki_route_value",
+            "rd_auto",
+            "timers_keepalive",
+        ]
+
+    def execute_module(self):
+        """ Execute the module
+
+        :rtype: A dictionary
+        :returns: The result from module execution
+        """
+        if self.state not in ["parsed", "gathered"]:
+            self.generate_commands()
+            self.run_commands()
+        return self.result
+
+    def generate_commands(self):
+        """ Generate configuration commands to send based on
+                    want, have and desired state.
+                """
+
+        for entry in self.want, self.have:
+            self._bgp_list_to_dict(entry)
+
+        # if state is deleted, clean up global params
+        if self.state == "deleted":
+            if not self.want or (
+                self.have.get("as_number") == self.want.get("as_number")
+            ):
+                self._compare(
+                    want={"as_number": self.want.get("as_number")},
+                    have=self.have,
+                )
+        elif self.state == "purged":
+            if not self.want or (
+                self.have.get("as_number") == self.want.get("as_number")
+            ):
+                self.addcmd(self.have or {}, "router", True)
+
+        else:
+            wantd = self.want
+            # if state is merged, merge want onto have and then compare
+            if self.state == "merged":
+                wantd = dict_merge(self.have, self.want)
+
+            self._compare(want=wantd, have=self.have)
+
+    def _compare(self, want, have):
+        """Leverages the base class `compare()` method and
+           populates the list of commands to be run by comparing
+           the `want` and `have` data with the `parsers` defined
+           for the Bgp_global network resource.
+        """
+
+        self.compare(parsers=self.parsers, want=want, have=have)
+        self._compare_rpki_server(want=want, have=self.have)
+        self._compare_confederation_peers(want=want, have=have)
+        self._compare_neighbors(want=want, have=self.have)
+        self._vrfs_compare(want=want, have=have)
+        if self.commands and "router bgp" not in self.commands[0]:
```

Comment:
iosxr_bgp_global:
   config:
      as_number: 65563
    state: merged
will this result in just `router_bgp 65563` ?

Reply:
yes